client/my-sites/domains/domain-management/site-redirect/index.jsx

```diff
@@ -3,151 +3,222 @@
-	},
+		this.closeRedirectNotice();
+	}
 
-	getInitialState() {
-		return {
-			location: this.props.location.value
-		};
-	},
+	closeRedirectNotice = () => {
+		upgradesActions.closeSiteRedirectNotice( this.props.selectedSite.domain );
+	};
 
-	handleChange( event ) {
+	handleChange = ( event ) => {
 		let location = event.target.value;
 
 		// Removes the protocol part
 		location = location.replace( /.*:\/\//, '' );
 
 		this.setState( { location } );
-	},
+	};
 
-	handleClick( event ) {
+	handleClick = ( event ) => {
 		event.preventDefault();
 
 		upgradesActions.updateSiteRedirect( this.props.selectedSite.domain, this.state.location, ( success ) => {
-			this.recordEvent( 'updateSiteRedirectClick', this.props.selectedDomainName, this.state.location, success );
+			this.props.updateSiteRedirectClick(
+				this.props.selectedDomainName,
+				this.state.location,
+				success
+			);
 
 			if ( success ) {
 				page( paths.domainManagementRedirectSettings( this.props.selectedSite.slug, this.state.location ) );
 			}
 		} );
-	},
+	};
 
-	handleFocus() {
-		this.recordEvent( 'locationFocus', this.props.selectedDomainName );
-	},
+	handleFocus = () => {
+		this.props.locationFocus( this.props.selectedDomainName );
+	};
 
 	render() {
-		let classes = classNames(
+		const { location, translate } = this.props;
+		const { isUpdating, notice } = location;
+		const isFetching = location.isFetching || this.state.location === null;
+
+		const classes = classNames(
 			'site-redirect-card',
-			{ fetching: this.props.location.isFetching }
+			{ fetching: isFetching }
 		);
 
 		return (
 			<div>
-				<SiteRedirectNotice
-					notice={ this.props.location.notice }
-					selectedSite={ this.props.selectedSite } />
-
-				<Main className="domain-management-site-redirect">
+				{
+					notice &&
+						<Notice
+							onDismissClick={ this.closeRedirectNotice }
+							status={ notices.getStatusHelper( notice ) }
+							text={ notice.text }
+						/>
+				}
+
+				<Main>
 					<Header onClick={ this.goToEdit } selectedDomainName={ this.props.selectedDomainName }>
-						{ this.translate( 'Redirect Settings' ) }
+						{ translate( 'Redirect Settings' ) }
 					</Header>
 
-					<SectionHeader label={ this.translate( 'Redirect Settings' ) } />
+					<SectionHeader label={ translate( 'Redirect Settings' ) } />
 
 					<Card className={ classes }>
 						<form>
 							<FormFieldset>
 								<FormLabel>
-									{ this.translate( 'Redirect To' ) }
+									{ translate( 'Redirect To' ) }
 								</FormLabel>
 
 								<FormTextInputWithAffixes
-									disabled={ this.props.location.isFetching || this.props.location.isUpdating }
+									disabled={ isFetching || isUpdating }
 									name="destination"
 									noWrap
 									onChange={ this.handleChange }
 									onFocus={ this.handleFocus }
 									prefix="http://"
 									type="text"
-									value={ this.state.location } />
+									value={ this.state.location || '' } />
```

Comment:
Problem is that then we loose the `null`. :) Well, in theory, empty state should also be invalid. Will try.